### PR TITLE
Update multiple Golang dependencies to latest version

### LINKED ISSUE #2177: Upgrade Operator-SDK to v1.27
### Requirement

We would like to update k8s apis.

### Problem

There are a few broken 
- https://github.com/jaegertracing/jaeger-operator/pull/2152
- https://github.com/jaegertracing/jaeger-operator/pull/2154
- https://github.com/jaegertracing/jaeger-operator/pull/2160
- https://github.com/jaegertracing/jaeger-operator/pull/2168
- https://github.com/jaegertracing/jaeger-operator/pull/2171

### Proposal

Updating Operator-sdk to the latest version.

### Open questions

_No response_

### PR DESCRIPTION
**Pull request by iblancasa**
## Which problem is this PR solving?
- Follow up for #2178
- Resolves #2177
- Closes #2171, #2168, #2160, #2154, #2152

### CODE REVIEW

File: pkg/metrics/instances.go [hunk 1]
```diff
@@ -162,12 +161,17 @@ func (i *instancesMetric) Setup(ctx context.Context) error {
...
 	for _, o := range i.observations {
-		instruments = append(instruments, o.Gauge)
+		_, err = meter.RegisterCallback(
+			func(ctx context.Context, observer metric.Observer) error {
+				return i.callback(ctx, observer)
+			},
+			o.Gauge)
+		if err != nil {
+			return err
+		}
```
Comment (frzifus) [inline pkg/metrics/instances.go:172]: Why do we need this custom callback here?
```suggestion
	instruments := make([]instrument.Asynchronous, 0, len(i.observations))
	for _, o := range i.observations {
		instruments = append(instruments, o.Gauge)
	}
	_, err = meter.RegisterCallback(i.callback, instruments...)
	return err
```

Comment (iblancasa) [reply pkg/metrics/instances.go:172]: The function signature changed and it was not building anymore.

Comment (frzifus) [reply pkg/metrics/instances.go:172]: yes, seems the swapped the parameters. For me its still unclear why you register multiple callbacks?

See:
https://github.com/open-telemetry/opentelemetry-go/pull/3587/commits/908d6e486afd4148a180cc17d1e85387c7a746b9#diff-600ae769b73b7ada58d7a7cfaf867d56778a91c8e36db76c5362670cdfb16e3eR284

Comment (iblancasa) [reply pkg/metrics/instances.go]: Changed

**Review by frzifus — APPROVED** [review]

**Pull request merged**